### PR TITLE
EDSC-2854: Creates limited shapefile for subsetting based on user selections

### PR DESCRIPTION
**Pull request by macrouch**

### CODE REVIEW

File: static/src/js/components/Map/ShapefileLayer.js [hunk 1]
```diff
@@ -105,10 +119,22 @@ class ShapefileLayerExtended extends L.Layer {
...
     // look through response and separate all MultiPolygon types into their own polygon
     this.separateMultiPolygons(response)
 
+    // Add an id field to each feature unless it already has one
+    response.features.map((feature, index) => {
+      const newFeature = feature
+      const { id } = feature
+
+      if (!id) {
+        newFeature.id = index.toString()
```
Comment (macrouch) [inline static/src/js/components/Map/ShapefileLayer.js:132]: change newFeature.id to newFeature.edscId to ensure we don't conflict with existing IDs

File: static/src/js/components/SpatialDisplay/SpatialDisplay.js [hunk 2]
```diff
@@ -734,6 +735,10 @@ class SpatialDisplay extends Component {
         </SpatialDisplayEntry>
       )
 
+      if (selectedFeatures.length) {
+        hint = `${selectedFeatures.length} ${pluralize('shape', selectedFeatures.length)} selected`
```
Comment (macrouch) [inline static/src/js/components/SpatialDisplay/SpatialDisplay.js:739]: if no selectedFeatures exist, hint should be 'Select a shape to filter results'

File: static/src/js/components/Facets/Facets.js [hunk 3]
```diff
@@ -115,6 +115,12 @@ const Facets = (props) => {
     autocompleteType: 'granule_data_format'
   }
 
+  const tilingSystem = {
```
Comment (abbottry) [inline static/src/js/components/Facets/Facets.js:118]: I don't think you want this here :)

File: migrations/1598036919806_add-parent-shapefile-id-selected-features-to-shapefiles.js [hunk 4]
```diff
@@ -0,0 +1,17 @@
+exports.shorthands = undefined
+
+exports.up = (pgm) => {
+  pgm.addColumns('shapefiles', {
+    parent_shapefile_id: {
+      type: 'integer',
+      references: 'shapefiles'
+    },
+    selected_features: {
+      type: 'text []'
```
[the remark this answers is not in the document]
Comment (macrouch) [reply migrations/1598036919806_add-parent-shapefile-id-selected-features-to-shapefiles.js:10]: Yep

**Pull request merged**